### DISCUSSION TITLE
How to get memory usage information after a Lambda function finishes running?

### DISCUSSION THREAD

**Discussion by holicc** [aws/aws-lambda-rust-runtime#935] [Q&A] [answered]
Is there any way we can obtain information about memory usage, similar to the CloudWatch log output below?

```text
REPORT RequestId: 75b41491-650d-4365-86f8-a0bd2842b6ec        Duration: 771.12 ms        Billed Duration: 772 ms        Memory Size: 10240 MB        Max Memory Used: 68 MB        
```
We need this metric value to report to our server.

**Comment by bnusunny** [comment, marked as answer]
You can enable [Lambda Insight ](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/Lambda-Insights.html)to get these Metrics.

If you don't want to use Lambda Insight, you can create a [Telemetry Extension](https://docs.aws.amazon.com/lambda/latest/dg/telemetry-api.html) to receive these metrics from Lambda and send to your server. 

The [platform.report](https://docs.aws.amazon.com/lambda/latest/dg/telemetry-schema-reference.html#platform-report) is the one you will get the data.
